submit_and_compare/submit_and_compare.py

```diff
@@ -266,32 +281,38 @@ def studio_submit(self, submissions, suffix=''):
 
     @XBlock.json_handler
     def send_hints(self, submissions, suffix=''):
-        tree = etree.parse(StringIO(self.question_string))
+        # pylint: disable=unused-argument
+        # pylint: disable=no-member
+        """
+        Build hints once for user
+        This is called once on page load and
+        js loop through hints on button click
+        """
+        tree = etree.parse(StringIO.StringIO(self.question_string))
         raw_hints = tree.xpath('/submit_and_compare/demandhint/hint')
-
         decorated_hints = list()
-
-        if len(raw_hints) == 1:
-            hint = u'Hint: ' + etree.tostring(raw_hints[0], encoding='unicode')
+        cnt = 0
```

Comment:
The preferred, Pythonic approach is to use the builtin `enumerate` call [1].

```python
for i, hint in enumerate(raw_hints):
    pass
```

[1] https://docs.python.org/2/library/functions.html#enumerate